File: day-1/typescript/1.ts
let leftList = []
let rightList = []
let diffSum = 0;

const text = await Deno.readTextFile("input-1.txt").then((text: string) => text.toString())

text.split("\n").forEach((line: string) => {
    const twoNumbers = line.split("   ");

    leftList.push(parseInt(twoNumbers[0]));
    rightList.push(parseInt(twoNumbers[1]));
});

leftList.sort((a, b) => a - b);
rightList.sort((a, b) => a - b);

for (let i = 0; i < leftList.length; i++) {
    diffSum += Math.abs(leftList[i] - rightList[i]);
}

console.log(diffSum)

File: day-1/python/1.py
left_list = []
right_list = []
diff_sum = 0

data = open("../input-1.txt", "r")

for line in data:
    left, right = line.split()

    left_list.append(int(left))
    right_list.append(int(right))

data.close()

for i in range(len(left_list)):
    diff_sum += abs(left_list[i] - right_list[i])

print(diff_sum)